diff --git a/src/index.js b/src/index.js
index a61c2d4500e881a73e99fe66853fb30c7f739a01..b1c45956bc02ce0380e2e89446f413d97210031c 100644
--- a/src/index.js
+++ b/src/index.js
@@ -210,6 +210,7 @@ export class SseTransport {
 		// Create new SSE stream
 		const stream = new ReadableStream({
 			start: (controller) => {
+				console.log(`Creating SSE stream for session ${session_id}`);
 				this.#options.sessionManager.create(session_id, controller);
 
 				// Send initial endpoint event with session info
@@ -399,6 +400,20 @@ export class SseTransport {
 			request.headers.get('mcp-session-id') ||
 			this.#options.getSessionId();
 
+		console.log("adding an abort listener and process listener for session", session_id);
+
+		process.on("SIGTERM", ()=>{
+			console.log(`SSE request for session ${session_id} aborted with SIGTERM`);
+			this.#options.sessionManager.delete(session_id);
+		})
+
+		request.signal.addEventListener('abort', () => {
+			console.log(`SSE request for session ${session_id} aborted`);
+			this.#options.sessionManager.delete(session_id);
+		});
+
+		console.log(`Handling SSE ${method} request for session ${session_id} ${url.searchParams.get('session_id') || request.headers.get('mcp-session-id')}`);
+
 		/**
 		 * @type {Response | null}
 		 */
